platform-core/platform-common/src/main/java/org/ekstep/common/PdfGenerator.java

```diff
@@ -0,0 +1,86 @@
+package org.ekstep.common;
+
+import com.itextpdf.text.Document;
+import com.itextpdf.text.pdf.PdfWriter;
+import com.itextpdf.tool.xml.XMLWorkerHelper;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class PdfGenerator {
+
+    private PdfGenerator() {
+
+    }
+
+    private static final String TEMP_FILE_LOCATION = Platform.config.hasPath("lp.assessment.tmp_file_location") ? Platform.config.getString("lp.assessment.tmp_file_location") : "/tmp/";
+
+    /**
+     * Converts a Html String to a Pdf file
+     *
+     * @param htmlString
+     * @return
+     */
+    public static File convertHtmlStringToPdfFile(String htmlString, String pdfFileName) {
+        try {
+            return convertFile(pdfFileName, getInputStream(htmlString));
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static File convertHtmlFileToPdfFile(File htmlFile, String pdfFileName) {
+        try {
+            return convertFile(pdfFileName, getInputStream(htmlFile));
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    private static File convertFile(String pdfFileName, InputStream is) throws Exception {
```

Comment:
dont pass Inputstream here. Pass file or String as Object and get Inputstream object in this method itself.

Reply:
Done